controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1334,38 +1334,45 @@ def test_unpublish_volume_get_host_by_host_identifiers_exception(self, storage_a
         self.servicer.ControllerUnpublishVolume(self.request, self.context)
-        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertEqual(self.context.code, status_code)
 
-        self.mediator.unmap_volume.side_effect = [array_errors.PermissionDeniedError("msg")]
-        storage_agent.return_value = self.storage_agent
-        self.servicer.ControllerUnpublishVolume(self.request, self.context)
-        self.assertEqual(self.context.code, grpc.StatusCode.PERMISSION_DENIED)
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_unpublish_volume_unmap_volume_object_not_found_error(self, storage_agent):
+        self._test_unpublish_volume_unmap_volume_with_error(storage_agent, array_errors.ObjectNotFoundError("volume"),
+                                                            grpc.StatusCode.OK)
 
-        self.mediator.unmap_volume.side_effect = [array_errors.HostNotFoundError("host")]
-        storage_agent.return_value = self.storage_agent
-        self.servicer.ControllerUnpublishVolume(self.request, self.context)
-        self.assertEqual(self.context.code, grpc.StatusCode.NOT_FOUND)
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_unpublish_volume_unmap_volume_volume_already_unmapped_error(self, storage_agent):
+        self._test_unpublish_volume_unmap_volume_with_error(storage_agent, array_errors.VolumeAlreadyUnmappedError(""),
+                                                            grpc.StatusCode.OK)
 
-        self.mediator.unmap_volume.side_effect = [array_errors.UnmappingError("", "", "")]
-        storage_agent.return_value = self.storage_agent
-        self.servicer.ControllerUnpublishVolume(self.request, self.context)
-        self.assertEqual(self.context.code, grpc.StatusCode.INTERNAL)
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_unpublish_volume_unmap_volume_permission_denied_error(self, storage_agent):
+        self._test_unpublish_volume_unmap_volume_with_error(storage_agent, array_errors.PermissionDeniedError("msg"),
+                                                            grpc.StatusCode.PERMISSION_DENIED)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_unpublish_volume_unmap_volume_host_not_found_error(self, storage_agent):
+        self._test_unpublish_volume_unmap_volume_with_error(storage_agent, array_errors.HostNotFoundError("host"),
+                                                            grpc.StatusCode.OK)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_unpublish_volume_unmap_volume_permission_denied_unmapping_error(self, storage_agent):
+        self._test_unpublish_volume_unmap_volume_with_error(storage_agent, array_errors.UnmappingError("", "", ""),
+                                                            grpc.StatusCode.INTERNAL)

```

Comment:
do we want the unit tests to enforce status codes that are not in the spec, for specific mediator exceptions?
if not at all, please remove this test.
if so, but not for specific mediator exceptions, please replace `UnmappingError` with e.g. `Exception`

Reply:
its not a new test, just separated.. and we are testing the code, not only the spec..

Reply:
ok, since it's not new - resolving